src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -0,0 +1,309 @@
+        /// </summary>
+        public string OperationName { get; }
+
+        /// <summary>
+        /// This is an ID that is specific to a particular request.   Filtering
+        /// to a particular ID insures that you get only one request that matches.  
+        /// It is typically assigned the system itself. 
+        /// </summary>
+        public string Id { get; private set; }
+
+        /// <summary>
+        /// The time that operation started.  Typcially when Start() is called 
+        /// (but you can pass a value to Start() if necessary.  This use UTC (Greenwitch Mean Time)
+        /// </summary>
+        public DateTime StartTimeUtc { get; private set; }
+
+        /// <summary>
+        /// If the Activity that created this activity is  from the same process you can get 
+        /// that Activit with Parent.   However this can be null if the Activity has no
+        /// parent (a root activity) or if the Parent is from outside the process.  (see ParentId for more)
+        /// </summary>
+        public Activity Parent { get; private set; }
+
+        /// <summary>
+        /// If the parent for this activity comes from outside the process, the activity
+        /// does not have a Parent Activity but MAY have a ParentId (which was serialized from
+        /// from the parent) .   This accessor fetches the parent ID if it exists at all.  
+        /// Note this can be null if this is a root Activity (it has no parent)
+        /// </summary>
+        public string ParentId { get; private set; }
+
+        /// <summary>
+        /// Tags are string-string key-value pairs that represent information that will
+        /// be logged along with the Activity to the logging system.   This information
+        /// however is NOT passed on to the children of this activity.  (see Baggage)
+        /// </summary>
+        public IEnumerable<KeyValuePair<string, string>> Tags
+        {
+            get
+            {
+                if (ParentId != null)
+                    yield return new KeyValuePair<string, string>("ParentId", ParentId);
+                for (var tags = _tags; tags != null; tags = tags.Next)
+                    yield return tags.keyValue;
+            }
+        }
+
+        /// <summary>
+        /// Tags are string-string key-value pairs that represent information that will
+        /// be passed along to children of this activity.   Baggage is serialized 
+        /// when requests leave the process (along with the ID).   Typically Baggage is
+        /// used to do fine-grained control over logging of the activty and any children.  
+        /// In general, if you are not using the data at runtime, you should be using Tags 
+        /// instead. 
+        /// </summary> 
+        public IEnumerable<KeyValuePair<string, string>> Baggage
+        {
+            get
+            {
+                for (var activity = this; activity != null; activity = activity.Parent)
```

Comment:
We would like to avoid copying baggage.  
Code which adds baggage to parent activity should expect that new child activity could be created/sent outside of process at any moment, so for this case there is no difference if Baggage/SpanContext is mutable or not: any child activity may get or may not get parent baggage item. 
There is a slight difference for in-proc children, that they will receive any parent update, but for the sake of performance, we allow it.